src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -4901,6 +4903,11 @@ public object VisitBinaryExpression(BinaryExpressionAst binaryExpressionAst)
                     return Expression.Call(CachedReflectionInfo.TypeOps_AsOperator, lhs.Cast(typeof(object)), rhs.Convert(typeof(Type)));
 
                 case TokenKind.DotDot:
+                    if(lhs.Type == typeof(string)){
```

Comment:
Will this work:
```
'A'..'Z' | % { ... }
```

Reply:
Yes, this will produce the sequence A (0x41) through Z (0x5A)

Reply:
Apparently it *does not work* on the Powershell Core 6.0.0 GA
```powershell
> 'A'..'Z' | % {$_ }
Cannot convert value "A" to type "System.Int32". Error: "Input string was not in a correct format."
At line:1 char:1
+ 'A'..'Z' | % {$_ }
+ ~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : InvalidArgument: (:) [], RuntimeException
+ FullyQualifiedErrorId : InvalidCastFromStringToInteger
```

Reply:
@kborowinski there is an open issue for that #5519

Reply:
@markekraus: sorry, I've missed that issue